src/main.rs

```diff
@@ -74,10 +75,15 @@ fn main() -> Result<i32> {
         true => Policy::limited(args.max_redirects.unwrap_or(10)),
         false => Policy::none(),
     };
+    let timeout = args
+        .timeout
+        .map(|t| t.as_duration())
+        .filter(|t| t != &Duration::from_nanos(0));
```

Comment:
Nice improvement!

I still think it would make a bit more sense to have this inside `as_duration()` but it doesn't really matter.